wagtail/core/models.py

```diff
@@ -299,6 +299,18 @@ def get_site_root_paths():
         """
         result = cache.get('wagtail_site_root_paths')
 
+        # Wagtail 2.11 changed the way site root paths were stored which would cause
+        # an upgraded 2.11 site to break when loading cached site root paths that were cached with 2.10.2
+        # or older versions of Wagtail.
+        # The code below checks if the any of the cached site urls is consistent with an older version of
+        # Wagtail and busts the cache.
+        if result is not None:
+            for site_root_paths in result:
+                if len(site_root_paths) == 3:
+                    cache.delete('wagtail_site_root_paths')
+                    result = None
+                    break
+
```

Comment:
Thanks @saevarom!

Just wondering - given that we're immediately setting the cache key again in the code below, deleting it is probably redundant... Could we achieve the same effect just by changing the line below to:

`if result is None or any(len(site_record == 3) for site_record in result):`

Reply:
You're right, this is cleaner. There is a slight parenthesis mismatch in your line so I will change it to:

`if result is None or any(len(site_record) == 3 for site_record in result):`

And commit a new change.